src/eos-updater-poll.c

```diff
@@ -17,90 +18,245 @@
  * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  * Boston, MA 02111-1307, USA.
  *
- * Author: Vivek Dasmohapatra <vivek@etla.org>
+ * Authors: Vivek Dasmohapatra <vivek@etla.org>
+ *          Krzesimir Nowak <krzesimir@kinvolk.io>
  */
 
-#include "eos-updater-poll.h"
+#include "config.h"
+
 #include "eos-updater-data.h"
+#include "eos-updater-poll-common.h"
+#include "eos-updater-poll-lan.h"
+#include "eos-updater-poll-main.h"
+#include "eos-updater-poll.h"
+
+#include "eos-util.h"
+
+#ifdef HAS_EOSMETRICS_0
+
+#include <eosmetrics/eosmetrics.h>
+
+#endif /* HAS_EOSMETRICS_0 */
+
+/*
+ * Records which branch will be used by the updater. The payload is a 4-tuple
+ * of 3 strings and boolean: vendor name, product ID, selected OStree ref, and
+ * whether the machine is on hold
+ */
+static const gchar *const EOS_UPDATER_BRANCH_SELECTED = "99f48aac-b5a0-426d-95f4-18af7d081c4e";
```

Comment:
Was there a reason to hardcode this? If it is a UUID, then perhaps that should be encoded in the name?